File: nodejs/data/codeSamples.js
/*
Copyright (c) 2016, Oracle and/or its affiliates. 
All rights reserved.
*/

module.exports.getCodeSample = getCodeSample = function (fileName, tag, callback) {
    var active = false,
        return_sample = '',
        rl = require('readline').createInterface({
        input: require('fs').createReadStream(fileName),
        terminal: false
    });

    rl.on('line', function (line) {
        if (active && line.indexOf('//SampleTagEnd') > -1){
            rl.close();
            active = false;
        } else if (active) {
            return_sample += line + '\n';
        } else {
            active = (line.indexOf('//SampleTagStart ' + tag) > -1)
        }
    });

    rl.on('close', function(){
        callback(null, return_sample);
    })
};

File: python/code_samples.py
'''
Copyright (c) 2016, Oracle and/or its affiliates. 
All rights reserved.
'''
def get_code_sample(file_name, tag):
    # openfile = open('members.py', 'r')
    active = False
    return_sample = ""
    print('#SampleTagStart ' + tag)
    with open(file_name) as f:
        for line in f:
            if '#SampleTagEnd ' + tag in line:
                break
            elif active:
                return_sample += line
            else:
                active = ('#SampleTagStart ' + tag in line)

    return return_sample